reason-mode.el

```diff
@@ -233,7 +233,7 @@ Argument END marks the end of the function."
   (setq-local parse-sexp-lookup-properties t))
 
 ;;;###autoload
-(add-to-list 'auto-mode-alist '("\\.rei?\\'" . reason-mode))
+(add-to-list 'auto-mode-alist '("\\.\\(res\\(i\\)?\\|rei\\)$" . reason-mode))
```

Comment:
```suggestion
(add-to-list 'auto-mode-alist '("\\.\\(resi?\\|rei?\\)$" . reason-mode))
```

```suggestion
(add-to-list 'auto-mode-alist '("\\.res?i?$" . reason-mode))
```

your commit disables the mode for re file if I undersand it correctly. And I think it can be simplified. But I haven't tested

Reply:
You're right, I updated the regex to support the commented files.